File: app/src/main/java/com/permissionnanny/TextDialogStubView.java
package com.permissionnanny;

import android.view.ViewStub;
import android.widget.TextView;
import butterknife.Bind;
import butterknife.ButterKnife;

/**
 *
 */
public class TextDialogStubView {

    private final ConfirmRequestBinder mBinder;

    @Bind(R.id.tvReason) TextView tvReason;

    public TextDialogStubView(ConfirmRequestBinder binder) {
        mBinder = binder;
    }

    public void inflateViewStub(ViewStub stub) {
        stub.setLayoutResource(R.layout.dialog_text);
        ButterKnife.bind(this, stub.inflate());
    }

    public void bindViews() {
        tvReason.setText(mBinder.getDialogBody());
    }
}

File: app/src/main/kotlin/com/permissionnanny/TextDialogStubView.kt
package com.permissionnanny

import android.view.ViewStub
import android.widget.TextView
import org.jetbrains.anko.find

/**

 */
open class TextDialogStubView(private val binder: ConfirmRequestBinder) {

    internal lateinit var tvReason: TextView

    open fun inflateViewStub(stub: ViewStub) {
        stub.layoutResource = R.layout.dialog_text
        val view = stub.inflate()
        tvReason = view.find(R.id.tvReason)
    }

    open fun bindViews() {
        tvReason.text = binder.dialogBody
    }
}
